### PR TITLE
replication mode: sync state to pd

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What is changed and how it works?
What's Changed:
Report dr autosync status to PD in heartbeats

### Check List 
Tests 
- Unit test

### Release note 
```release-note
None
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -197,8 +197,8 @@ procinfo = { git = "https://github.com/tikv/procinfo-rs", rev = "5125fc1a69496b7
 # When you modify TiKV cooperatively with kvproto, this will be useful to submit the PR to TiKV and the PR to
 # kvproto at the same time.
 # After the PR to kvproto is merged, remember to comment this out and run `cargo update -p kvproto`.
-# [patch.'https://github.com/pingcap/kvproto']
-# kvproto = {git = "https://github.com/your_github_id/kvproto", branch="your_branch"}
+[patch.'https://github.com/pingcap/kvproto']
+kvproto = {git = "https://github.com/oh-my-tidb/kvproto", branch="async-wait"}
```
Comment (nolouch) [inline Cargo.toml:201]: need update kvproto

**Review by nolouch — COMMENTED** [review]
rest LGTM

File: tests/integrations/pd/test_rpc_client.rs [hunk 2]
```diff
@@ -122,7 +122,12 @@ fn test_rpc_client() {
     assert_eq!(region_info.region, region);
     assert_eq!(region_info.leader.unwrap(), peer);
 
-    block_on(client.store_heartbeat(pdpb::StoreStats::default(), /*store_report=*/ None)).unwrap();
+    block_on(client.store_heartbeat(
+        pdpb::StoreStats::default(),
+        /*store_report=*/ None,
```
Comment (nolouch) [inline tests/integrations/pd/test_rpc_client.rs:127]: Do we need to add a test to verify the state?

**Review by lhy1024 — APPROVED** [review]
LGTM

**Review by nolouch — APPROVED** [review]
lgtm

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**